src/veh_interact.cpp

```diff
@@ -371,6 +378,15 @@ void veh_interact::cache_tool_availability()
                 crafting_inv.has_components( "wheel_bicycle", 1 ) ||
                 crafting_inv.has_components( "wheel_motorbike", 1 ) ||
                 crafting_inv.has_components( "wheel_small", 1 );
+
+    int qual_jack = ceil( double( veh->total_mass() * 1000 ) / TOOL_LIFT_FACTOR );
+    has_jack = g->u.has_quality( "JACK", qual_jack ) ||
+               map_selector( g->u.pos(), PICKUP_RANGE ).has_quality( "JACK", qual_jack ) ||
+               vehicle_selector( g->u.pos(), 1, *veh ).has_quality( "JACK", qual_jack );
```

Comment:
> Then why limit them to 1 tile range?

Consistent behavior in all circumstances -  cranes, forklift trucks and digger buckets should all have similar semantics. I'd like to model a realistic set of construction vehicles for works depots and the like. It doesn't make any sense why most mechanisms would work from 20 tiles away.

> Having to haul the crane around would be pure annoyance.

The folding engine crane is sufficient for almost all parts installation tasks. If you're working on a sufficiently large vehicle you're already going to have to haul tools around due to `PICKUP_RANGE`.

> If you need a separate vehicle to operate on main vehicle, you're going to build the simplest one possible.

Sure, but in another scenario you might also be using your main vehicle to salvage parts from another (much larger) vehicle/wreckage. Further ancillary uses are likely to develop with time.

> Protrusion won't make it vulnerable because it will be deconstructed when not used, not driven around.

Adding/removing parts needs to have a non-trivial time penalty.